src/drivers/SVG/Fl_SVG_File_Surface.cxx

```diff
@@ -205,13 +207,13 @@ void Fl_SVG_Graphics_Driver::compute_dasharray(float s, char *dashes) {
       sprintf(dasharray_+strlen(dasharray_), "%.3f,", (*p)/s);
     }
     dasharray_[strlen(dasharray_) - 1] = 0;
-    if (user_dash_array_ != dashes) user_dash_array_ = strdup(dashes);
+    if (user_dash_array_ != dashes) user_dash_array_ = fl_strdup(dashes);
     return;
   }
   int dash_part = line_style_ & 0xFF;
   if (dash_part == FL_SOLID)  {
     if (dasharray_ && strcmp(dasharray_, "none")) free(dasharray_);
-    dasharray_ = strdup("none");
+    dasharray_ = fl_strdup("none");
```

Comment:
Just a side note: I'm not sure what's going on here, but perhaps someone should double-check if this can leak memory by skipping the `free()`.

Reply:
Good catch! I believe we need either:
```
    if (dasharray_ && strcmp(dasharray_, "none")) free(dasharray_);
    dasharray_ = "none";
```
which would skip the `free` because "none" is not (never) `strdup()`'ed or
```
    if (dasharray_) free(dasharray_);
    dasharray_ = fl_strdup("none");
```
which would `strdup()` "none" and free it the next time.

Is this what you meant?
If yes, this is something @ManoloFLTK should take a look at (it's his code), but I'm pretty sure you found a potential leak.

Reply:
There's no leak there. Please, consider all instances of the dasharray_ member variable.

Reply:
Hmm, in the ctor initializes `dasharray_ = strdup("none");`, so if this `( dash_part == FL_SOLID)` path happens while it's set to "none", it'll leak, right? e.g.

```
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    // Init
    char *dasharray_ = strdup("none");
    // Stuff happens..
    if ( dasharray_ && strcmp(dasharray_, "none"))
      { printf("free '%s'\n"); free(dasharray_); dasharray_ = 0; } // [A]
    if ( dasharray_ ) printf("ABOUT TO LEAK MEMORY\n");
    dasharray_ = strdup("none");  // [B]
    return 0;
}
```
..this prints 'ABOUT TO LEAK MEMORY' because it's about to overwrite dash_array_ with a new strdup() copy while it's still set to the ctor initialized value.

It just seems scary to have a condition on the free() [A], but not on the strdup() assignment at [B] right after it.

Reply:
Sorry. There's a memory leak. Closed now.

Reply:
FTR Manolo's fix applied as 5b1c7249a5bc219f01ebc3e9b9cf64b678039de8